packages/perseus/src/widget-type-utils.test.ts

```diff
@@ -0,0 +1,271 @@
+import {
+    generateTestCategorizerWidget,
+    generateTestPerseusItem,
+    generateTestRadioWidget,
+} from "./util/test-utils";
+import {
+    getWidgetTypeByWidgetId,
+    contentHasWidgetType,
+    getWidgetIdsFromContent,
+    getWidgetIdsFromContentByType,
+    getWidgetsMapFromItemData,
+    getWidgetFromWidgetMap,
+    getWidgetsFromWidgetMap,
+} from "./widget-type-utils";
+
+describe("widget-type-utils", () => {
+    describe("getWidgetTypeByWidgetKey", () => {
+        it("returns widget type when found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBe("radio");
+        });
+
+        it("returns null when not found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {};
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBeNull();
+        });
+    });
+
+    describe("contentHasWidgetType", () => {
+        it("returns true when found", () => {
+            // Assemble
+            const type = "radio";
+            const widgetId = "dont-look-for-type-in-key";
```

Comment:
Suggestion, nb: Oh, just noticed that all these IDs refer to themselves as keys. Should it use "id"?
```suggestion
            const widgetId = "dont-look-for-type-in-id";
```